nx/lib/nx.ex

```diff
@@ -375,7 +375,8 @@ defmodule Nx do
   See also: `t:t/0`
   """
   @doc type: :guards
-  defguard is_tensor(t) when is_number(t) or is_struct(t, T) or is_struct(t, Complex)
+  defguard is_tensor(t)
+           when is_number(t) or is_struct(t, T) or is_struct(t, Complex) or is_boolean(t)
```

Comment:
I don't think we should do this change (for the same reason we don't include `is_list(t)` here).